contrib/IECoreAppleseed/src/IECoreAppleseed/outputDriver/DisplayTileCallback.cpp

```diff
@@ -226,131 +240,100 @@ class DisplayLayer
 		asf::Image *m_image;
-		Box2i m_data_window;
+		Box2i m_dataWindow;
 		vector<float> m_buffer;
-		size_t m_tile_width;
-		size_t m_tile_height;
-		size_t m_channel_count;
+		size_t m_tileWidth;
+		size_t m_tileHeight;
+		size_t m_channelCount;
 		string m_layerName;
 		asf::Dictionary m_params;
 
 };
 
+using DisplayLayers = vector<unique_ptr<DisplayLayer>>;
+using DisplayLayersPtr = shared_ptr<DisplayLayers>;
+
 class DisplayTileCallback : public ProgressTileCallback
 {
 
 	public :
 
-		explicit DisplayTileCallback( const asr::ParamArray &params ) : ProgressTileCallback(), m_displays_initialized( false )
+		DisplayTileCallback( const DisplayLayersPtr &layers )
+			:	m_layers( layers )
 		{
-			// Create display layers.
-			m_layers.reserve( params.dictionaries().size() );
-			for( auto it( params.dictionaries().begin() ), eIt( params.dictionaries().end() ); it != eIt; ++it )
-			{
-				m_layers.push_back( new DisplayLayer( it.key(), it.value() ) );
-			}
 		}
 
 		~DisplayTileCallback() override
 		{
-			for( DisplayLayer *layer : m_layers )
-			{
-				delete layer;
-			}
 		}
 
 		void release() override
 		{
-			// We don't need to do anything here.
-			// The tile callback factory deletes this instance.
+			delete this;
 		}
 
 		void on_tile_begin(const asr::Frame *frame, const size_t tileX, const size_t tileY) override
 		{
 			const asf::CanvasProperties &props = frame->image().properties();
 			const size_t x = tileX * props.m_tile_width;
 			const size_t y = tileY * props.m_tile_height;
-
-			boost::lock_guard<boost::mutex> guard( m_mutex );
-
-			if( m_displays_initialized )
+			for( auto &layer : *m_layers )
 			{
-				for( DisplayLayer *layer : m_layers )
-				{
-					layer->hightlight_region( x, y, props.m_tile_width, props.m_tile_height );
-				}
+				layer->initDisplay( frame );
+				layer->highlightRegion( x, y, props.m_tile_width, props.m_tile_height );
 			}
 		}
 
 		void on_tile_end(const asr::Frame *frame, const size_t tileX, const size_t tileY) override
 		{
-			boost::lock_guard<boost::mutex> guard( m_mutex );
-
-			init_layer_displays( frame );
+			ProgressTileCallback::on_tile_end( frame, tileX, tileY );
 
-			for( DisplayLayer *layer : m_layers )
+			for( auto &layer : *m_layers )
 			{
-				layer->write_tile( tileX, tileY );
+				layer->initDisplay( frame );
+				layer->writeTile( tileX, tileY );
 			}
-
-			log_progress( frame, tileX, tileY );
 		}
 
 		void on_progressive_frame_update( const asr::Frame* frame ) override
 		{
-			boost::lock_guard<boost::mutex> guard( m_mutex );
-
-			init_layer_displays( frame );
-
 			const asf::CanvasProperties &frame_props = frame->image().properties();
 
 			for( size_t ty = 0; ty < frame_props.m_tile_count_y; ++ty )
 			{
 				for( size_t tx = 0; tx < frame_props.m_tile_count_x; ++tx )
 				{
-					for( DisplayLayer *layer : m_layers )
+					for( auto &layer : *m_layers )
 					{
-						layer->write_tile( tx, ty );
+						layer->initDisplay( frame );
```

Comment:
Same here, where the optimization might be somewhat more relevant given the potential number of tiles.

Reply:
Done in https://github.com/ImageEngine/cortex/pull/1011/commits/b9a61c17d37a2b2492f8f1868b2cbb9c59877c19